samples/nrf9160/modem_shell/src/cloud/cloud_mqtt_shell.c

```diff
@@ -240,13 +241,21 @@ static void cmd_cloud_connect(const struct shell *shell, size_t argc, char **arg
 
 static void cmd_cloud_disconnect(const struct shell *shell, size_t argc, char **argv)
 {
-	int err = nrf_cloud_disconnect();
+	int err;
 
+	if (!initialized) {
+		mosh_print("Not connected to nRF Cloud");
+		return;
+	}
```

Comment:
Do we really need this branch? It's only entered before the initial `cloud connect` command.
We could call `k_work_cancel_delayable` and `nrf_cloud_disconnect` if somebody wants to call `cloud disconnect` before `cloud connect`.

Reply:
The reason for this is that the delayable work only gets initialized when `cmd_cloud_connect()` is called. If user gives command `cloud disconnect` without first doing `cloud connect` the delayable work will be uninitialized when `k_work_cancel_delayable()` is called for it.

I could change the initialization so, that the work items are initialized at compile-time using `K_WORK_DEFINE` and `K_DELAYED_WORK_DEFINE` macros instead. Then we can get rid of that branch.

Reply:
> If user gives command `cloud disconnect` without first doing `cloud connect` the delayable work will be uninitialized when `k_work_cancel_delayable()` is called for it.

This didn't seem to cause issues but documentation doesn't state anything so would be questionable to do that.

> I could change the initialization so, that the work items are initialized at compile-time using `K_WORK_DEFINE` and `K_DELAYED_WORK_DEFINE` macros instead. Then we can get rid of that branch.

This is one option although I started to get bothered also about nrf_cloud_init() call and why it's here. Some modules does this by providing `init()` function to MoSh main.c. It seems this cloud mqtt and GNSS modules have initialization done during first use of the module and protected by a static variable.

I'd expect that using macros and running initialization would reserve some more memory which would be unnecessary if `cloud connect` is never used.

Both approaches have pros and cons so I'm not sure we should change it. Up to you.

Reply:
Changing the initialization might affect memory consumption (when these features are not used), but I'm more afraid that initializing the nRF Cloud library during boot breaks something because it sets the mem hooks. I have now changed the work item initialization and removed that one `if` branch.